tests/CMakeLists.txt

```diff
@@ -1,25 +1,27 @@
+function(traccc_add_test name)
+    # Parse the function's options.
+    cmake_parse_arguments(ARG "" "" "SOURCES;LINK_LIBRARIES" ${ARGN})
 
-macro(add_traccc_test TESTNAME FILES PLUGIN_LIBRARY)
-    add_executable(${TESTNAME} ${FILES})
-    target_link_libraries(${TESTNAME} gtest gmock gtest_main)
-    target_link_libraries(${TESTNAME} traccc::core)
-    target_link_libraries(${TESTNAME} traccc::tests::common)
-    target_link_libraries(${TESTNAME} ${PLUGIN_LIBRARY})
-    set_target_properties(${TESTNAME} PROPERTIES FOLDER tests)
-    add_test(NAME unit_test_${TESTNAME} COMMAND ${TESTNAME})
-    set_tests_properties(unit_test_${TESTNAME} PROPERTIES
-    ENVIRONMENT TRACCC_TEST_DATA_DIR=${PROJECT_SOURCE_DIR}/data/)
-endmacro()
+    # Create the test executable.
+    set(TEST_NAME "test_${name}")
+    add_executable(${TEST_NAME} ${ARG_SOURCES})
```

Comment:
For me it's still very much here... :confused:

Reply:
I think I'm going mad.